### DISCUSSION TITLE
Failure during make x11iraf on macOS Catalina

### DISCUSSION THREAD

**Discussion by tsumutsumu7** [iraf-community/iraf#228] [Q&A] [answered]
I'm getting this error when I try to make x11iraf.
Any idea on how to fix this problem?
Thanks.
```
cc -L/usr/X11/lib -o xgterm button.o charproc.o cursor.o data.o gtermio.o input.o main.o menu.o misc.o screen.o scrollbar.o tabs.o util.o VTPrsTbl.o \
	    -L../obm -lobm \
	    -lXpm -ltcl -lXaw3d -lXmu -lXt -lXext -lX11 -lncurses -lm
Undefined symbols for architecture x86_64:
  "_Tcl_GetErrorLine", referenced from:
      _ObmDeliverMsg in libobm.a(Obm.o)
      _obm_call_activate_callbacks in libobm.a(Obm.o)
      _gtermInputCallback in libobm.a(gterm.o)
      _gtermResizeCallback in libobm.a(gterm.o)
      _gtermResetCallback in libobm.a(gterm.o)
      _cb_error in libobm.a(html.o)
      _MarkerEvaluate in libobm.a(marker.o)
      ...
  "_Tcl_SetErrorLine", referenced from:
      _MarkerEvaluate in libobm.a(marker.o)
      _ParameterEvaluate in libobm.a(param.o)
      _WidgetEvaluate in libobm.a(widget.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [xgterm] Error 1
make: *** [xgterm] Error 2
```

**Comment by olebole** [comment, marked as answer]
This is probably the same problem as in iraf-community/x11iraf#46: for some reason, the "brew" installs a newer `tcl.h` on `/usr/local/include`, where it is found before the system's one `/usr/include/tcl.h`. This newer header relies on a newer libtcl, which is however not in the path. There are two ways to fix this: Either you remove `/usr/local/include/tcl.h` (or move if somewhere else), or add the path of the corresponding libtcl to `LD_LIBRARY_PATH`. This is a bug in brew, which should be reported there.

**Reply by tsumutsumu7** [reply]
I just took a peek into my system.
/usr/local/include/tcl.h
was there.
However
/usr/include/
was not in the directory itself.

**Reply by tsumutsumu7** [reply]
Do I just move the '/usr/local/include/tcl.h' to /usr/include/ ?

**Reply by olebole** [reply]
No, there is already one in /use/include. This one is the one to be used.

**Reply by tsumutsumu7** [reply]
No /usr/include/ as shown below.

~ % ls /usr/
X11		appleinternal	lib		local		share
X11R6		bin		libexec		sbin		standalone

**Reply by tsumutsumu7** [reply]
I moved '/usr/local/include/tcl.h' to ~/ and tried make, but the error did not change.

**Reply by olebole** [reply]
You should move this `tcl.h` out of the defalt searchpath, or rename it, before compiling. And do a `make clean` before to ensure the sources are recompiled.

**Reply by tsumutsumu7** [reply]
Now move '/usr/local/include/tcl.h' to ~/tcl.h.old.
I ran 'make clean' and then 'make'.
The xgterm and ximtool executable files were created.

Thank you very much.
